cellengine/resources/gate.py

```diff
@@ -193,51 +96,197 @@ def create(
         locked: bool = False,
         parent_population_id: str = None,
-        parent_population: str = None,
+        parent_population: str = None,  # TODO
         tailored_per_file: bool = False,
         fcs_file_id: str = None,
         fcs_file: str = None,
         create_population: bool = True,
     ) -> RectangleGate:
-        g = format_rectangle_gate(**get_args_as_kwargs(cls, locals()))
-        return cls(g)
+        """Formats a rectangle gate for posting to the CellEngine API.
 
+        Args:
+            experiment_id (str): The ID of the experiment.
+            x_channel (float): The name of the x channel to which the gate applies.
+            y_channel (float): The name of the y channel to which the gate applies.
+            name (str): The name of the gate x1 (float): The first x coordinate
+                (after the channel's scale has been applied).
+            x2 (float): The second x coordinate (after the channel's scale
+                has been applied).
+            y1 (float): The first y coordinate (after the channel's scale has
+                been applied).
+            y2 (float): The second y coordinate (after the channel's scale has
+                been applied).
+            label (float): Position of the label. Defaults to the midpoint of the gate.
+            gid (str): Group ID of the gate, used for tailoring. If this is not
+                specified, then a new Group ID will be created. To create a
+                tailored gate, the gid of the global tailored gate must be specified.
+            locked (bool): Prevents modification of the gate via the web interface.
+            parent_population_id (str): ID of the parent population. Use ``None`` for
+                the "ungated" population. If specified, do not specify
+                ``parent_population``.
+            parent_population (str): Name of the parent population. An attempt will
+                be made to find the population by name.  If zero or more than
+                one population exists with the name, an error will be thrown.
+                If specified, do not specify ``parent_population_id``.
+            tailored_per_file (bool): Whether or not this gate is tailored per FCS file.
+            fcs_file_id (str): ID of FCS file, if tailored per file. Use ``None`` for
+                the global gate in a tailored gate group. If specified, do not
+                specify ``fcs_file``.
+            fcs_file (str): Name of FCS file, if tailored per file. An attempt
+                will be made to find the file by name. If zero or more than one file
+                exists with the name, an error will be thrown. Looking up files
+                by name is slower than using the ID, as this requires
+                additional requests to the server. If specified, do not specify
+                ``fcs_file_id``.
+            create_population (bool): Automatically create corresponding population.
```

Comment:
2/

And not be an arg here since it's not a resource property and since this doesn't make an API request.